### ISSUE TITLE
onnxruntime XGBoost prediction score mismatch with n_estimators > 1

### ISSUE THREAD

**Issue by shirazzaman** [onnx/onnx#2228]
I have a simple model with 7 float columns.
When number of trees == 1, onnxruntime and xgboost predict exact scores.
However, when number of trees is more than 1, onnxruntime overpredicts the value.

For example, 
when n_estimators == 1 
    then xgboost predicts '0.4975429' and onnxruntime predicts '0.4975429'
when n_estimators == 2
   then xgboost predicts '0.4967465' and onnxruntime predicts '0.5023072'

**Comment by ebarsoum** [comment]
Can you open an issue in ONNX Runtime (https://github.com/microsoft/onnxruntime )?

**Comment by prabhat00155** [comment]
I can't reproduce this problem.

X, y = make_classification(n_features=40, n_samples=100, random_state=42, n_classes=5, n_informative=7)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=42)

model = XGBClassifier(random_state=42, learning_rate=.05, n_estimators=6)
model.fit(X_train, y_train)

onnx_fs = convert_xgboost(model, 'xgb', [('float_input', FloatTensorType([1, X_test.shape[1]]))])
save_model(onnx_fs, 'xgb.onnx')
sess = InferenceSession('xgb.onnx')
res = sess.run('', input_feed={'float_input': X_test.astype(np.float32)})
print(np.mean(np.isclose(model.predict_proba(X_test), res[1], atol=1e-7)))
print(np.mean(res[0] == np.argmax(model.predict_proba(X_test), axis=1)))

output:
1.0
1.0 

Could you share your script?

**Comment by shirazzaman** [comment]
Try Regressor or Classifier many times. Binary classification would match
50% of the time.

**Comment by prabhat00155** [comment]
I see the problem here, the converter just returns the index of the class(starting from 0) rather than the actual class itself. In the above example, onnxruntime returns 6, which is actually the index for class 7, which you can see if you print model.classes_. This needs to be fixed of course. I'll work on it. This seems to be a different issue from what you had originally reported: mismatches with n_estimators > 1 for both classifiers and regressors. Can you give examples for that as well?

**Comment by prabhat00155** [comment]
https://github.com/onnx/onnxmltools/pull/336 fixes the label mismatch issue I explained above.
